solr/solr-ref-guide/src/neural-search.adoc

```diff
@@ -0,0 +1,320 @@
+= Neural Search
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+Search comprises of performing four primary steps:
+
+* generate a representation of the query that specifies the information need
+* generate a representation of the document that captures the information contained
+* match the query and the document representations from the corpus of information
+* assign a score to each matched document in order to establish a meaningful document ranking by relevance in the results
+
+With the *Neural Search* module Apache Solr is introducing support for neural networks based techniques that can improve various aspects of search.
```

Comment:
```suggestion
With the *Neural Search* module Apache Solr is support for neural networks based techniques that can improve various aspects of search.
```

Reply:
I am not a native speaker, but is "With the *Neural Search* module Apache Solr is support for neural networks based techniques" correct? I didn't know of the construct:
"with <something> Apache Solr is support for <something>

Reply:
```suggestion
The Apache Solr *Neural Search* module adds support for neural networks based techniques that can improve various aspects of search.
```

Reworded again to maybe be a bit more clear and direct.